Samples/ControlMonitorApp/ControlMonitorHelper/ControlMonitorHelper.cpp

```diff
@@ -0,0 +1,150 @@
+//
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//
+
+#include "pch.h"
+#include "ControlMonitorHelper.h"
+#include "ControlMonitorManager.g.cpp"
+
+namespace winrt::ControlMonitorHelper::implementation
+{
+    winrt::ControlMonitorHelper::ControlMonitorManager ControlMonitorManager::CreateCameraControlMonitor(winrt::hstring symlink, winrt::array_view<winrt::ControlMonitorHelper::ControlData const> controls)
+    {
+        auto result = winrt::make<ControlMonitorManager>(symlink, controls);
+        return result;
+    }
+
+    ControlMonitorManager::ControlMonitorManager(winrt::hstring cameraSymbolicLink, array_view<winrt::ControlMonitorHelper::ControlData const> controls)
+    {
+        m_spCallback = winrt::make_self<CameraControlCallback>(this);
+
+        HRESULT hr = MFCreateCameraControlMonitor(cameraSymbolicLink.c_str(), m_spCallback.get(), &m_spMonitor);
+        if (FAILED(hr))
+        {
+            throw winrt::hresult_error(hr, L"error from MFCreateCameraControlMonitor..");
+        }
+
+        m_controls = controls;
+
+        for each (auto&& var in m_controls)
+        {
+            GUID guid{};
+            switch (var.controlKind)
+            {
+            case ControlKind::VidCapCameraControlKind:
+                guid = PROPSETID_VIDCAP_CAMERACONTROL;
+                break;
+            case ControlKind::VidCapVideoProcAmpKind:
+                guid = PROPSETID_VIDCAP_VIDEOPROCAMP;
+                break;
+            case ControlKind::ExtendedControlKind:
+                guid = KSPROPERTYSETID_ExtendedCameraControl;
+                break;
+            default:
+                throw winrt::hresult_error(E_UNEXPECTED, L"Invalid ControlKind");
+            }
+
+            hr = m_spMonitor->AddControlSubscription(guid, var.controlId);
+            if (FAILED(hr))
+            {
+                throw winrt::hresult_error(hr, L"error from IMFCameraControlMonitor::AddControlSubscription(PROPSETID_VIDCAP_VIDEOPROCAMP, 0)..");
+            }
+        }
+
+        hr = m_spMonitor->Start();
+        if (FAILED(hr))
+        {
+            throw winrt::hresult_error(hr, L"error from Start IMFCameraControlMonitor..");
+        }
+    }
+
+    ControlMonitorManager::~ControlMonitorManager() noexcept try
+    {
+        if (m_spMonitor != nullptr)
+        {
+            THROW_IF_FAILED(m_spMonitor->Stop());
+            for each (auto && var in m_controls)
+            {
+                GUID guid{};
+                switch (var.controlKind)

```

Comment:
hmm.. why is this there and why is line 84 just using hardcoded definition of what to remove subscription for?

Reply:
Good point, missed this while modifying the codebase from Michael to support more than just one hard coded control.